feat: Complete mobile UI/UX overhaul for some questionnaire components

yash-learner opened the pull request:
## Proposed Changes

- Fixes #11110 
- a comprehensive mobile UI/UX overhaul 

This pull request makes significant improvements to the mobile UI/UX for questionnaire components in the care_fe application. Key changes include:

# TLDR

- Added sheet-based navigation for adding a new `MedicationRequest`, `MedicationStatement`, `Allergy`, which uses extracted [src/components/Questionnaire/EntitySelectionDrawer.tsx](https://github.com/ohcnetwork/care_fe/pull/11642/files#diff-c1784f9dfa1c54fd2256fcf8c8feecb3c89cc054f6efb48fa75b40142061d34cl)
- Used  extracted [src/components/Questionnaire/EntitySelectionDrawer.tsx](https://github.com/ohcnetwork/care_fe/pull/11642/files#diff-c1784f9dfa1c54fd2256fcf8c8feecb3c89cc054f6efb48fa75b40142061d34cl) and updated the `Symptom` and `diagnosis`
- Updated the `NoteInput` used across the questionnaire from popover to inline `text area` with a toggle.
- Added a[ MultiValueSetSelec](https://github.com/ohcnetwork/care_fe/pull/11642/files#diff-e87e8517e46375459daf0441b9676da0f2937de3edfaa18207553455c755efa1)t  with the sheet for making it mobile friendly.
- ~~Created a mobile friendly scrollable [component for dosag](https://github.com/ohcnetwork/care_fe/pull/11642/commits/9d75cf995619d10682ac161a25aa8e4d1683a18d#diff-6076f7fb9bd5dd9eb59d964c26979edc379c6dd4f3e34a39a1ef2dfebc981ccb)e input selection.~~ Reverted this as popovers started working after react 19 upgrade.
- Lot of tiny UI ehancements.

----

## Localization
- **public/locale/en.json**: Added new translation strings to support mobile UI components and entity selection workflows

## Medicine Components
- **src/components/Medicine/MedicationAdministration/MedicineAdminForm.tsx**: 
  - Added proper object handling for time units similar to dosage
  - Updated code to send proper codes to backend instead of objects

## Patient Components
- **src/components/Patient/allergy/list.tsx**:
  - Improved mobile display with collapsible sections
  - Enhanced readability with proper spacing and card components

## Questionnaire Components
- **src/components/Questionnaire/EntitySelectionDrawer.tsx**:
  - Created new sheet-based UI for entity selection on mobile
  - Added proper header, navigation controls, and overflow handling
  - Used dvh units for viewport height consistency across devices
  - Implemented back navigation and confirmation patterns

- **src/components/Questionnaire/ValueSetSelect.tsx**:
  - Added title and onBack props for better mobile navigation
  - Enhanced search functionality and improved display of selected items

## Question Types
- **AllergyQuestion.tsx**: 
  - Added mobile drawer support for allergy selection
  - Improved allergy display and management UI

- **ChoiceQuestion.tsx**:
  - Refactored for consistent button sizing and mobile-friendly layout
  - Improved spacing and touch targets

- **DiagnosisQuestion.tsx**:
  - Enhanced mobile UI with proper card components
  - Improved diagnosis selection workflow

- **MedicationRequestQuestion.tsx** & **MedicationStatementQuestion.tsx**:
  - Added drawer-based medication selection for mobile
  - Improved medication status display
  - Updated grid layout for better mobile viewing

- **NoteToggle.tsx** & **NotesInput.tsx**:
  - Kept notes inline with improved UI/UX
  - Enhanced mobile experience with better input controls

- **QuestionInput.tsx**:
  - Refactored for consistent button sizing across question types
  - Improved layout responsiveness

- **SymptomQuestion.tsx** & **TextQuestion.tsx**:
  - Enhanced mobile UI with better spacing and layout
  - Improved input controls and selection interfaces

## Types
- **src/types/emr/medicationRequest.ts**:
  - Updated types to support enhanced medication request handling

Throughout all files, improvements include proper Tailwind sizing classes, enhanced spacing, better touch targets, and consistent UI patterns optimized for mobile screens.

   
 ---- summary -----
 
  - Introduced a mobile-friendly selection drawer for easier configuration of medications, allergies, medication request, and medication statement.
  - Added interactive note management for toggling and editing notes.
  - Updated user prompts with clearer calls-to-action (e.g., “Add Note…”, “Enter reason for Medication”).

- **Style**
  - Enhanced input styling with refined textarea, rounded borders, and updated icon sizes for improved visual consistency.
  - Improved mobile navigation with clearer back actions and responsive layouts.



@ohcnetwork/care-fe-code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [X] Ensure that UI text is kept in I18n files.
- [X] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [X] Request for Peer Reviews
- [ ] Completion of QA in Mobile Devices
- [ ] Completion of QA in Desktop Devices


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Introduced a mobile-friendly drawer interface for selecting and configuring medical entities (medications, allergies, symptoms, diagnoses).
  - Added a new toggle for viewing and adding notes, with improved note management in questionnaires.
  - Enhanced support for time unit selection and display (years, months, weeks, days, hours) with localization.

- **Improvements**
  - Refined mobile layouts for medication, allergy, symptom, and diagnosis selection and editing.
  - Updated UI components for better visual structure and responsiveness, including larger allergy icons and improved card/collapsible layouts.
  - Improved medication duration input and unit selection with stricter typing and display logic.
  - Simplified and unified note input experience with dynamic textarea display.
  - Enhanced styling of select triggers and textarea elements for consistency.
  - Added optional titles to selection components for better context.

- **Bug Fixes**
  - Fixed styling inconsistencies in select triggers and textarea elements.

- **Chores**
  - Updated and reorganized localization keys for clarity and expanded pluralization support.
  - Removed deprecated note input component and unused code.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->



https://github.com/user-attachments/assets/71aab1ab-5566-4709-92b1-ea12da630af0

yash-learner committed:
Add card and collapsible

yash-learner committed:
Merge branch 'develop' into improve-ux-other-qc

yash-learner committed:
Render separate styling for valusets in Advise Medicine

yash-learner committed:
Refactor: use a proper object for time units and similar to dosage and update relevant code

yash-learner committed:
Send the code to BE not object

yash-learner committed:
Improve UI/UX using collapsible and card

yash-learner committed:
Remove pointer-events-none

yash-learner committed:
Update localization strings and improve medication status display in Questionnaire and move note

yash-learner committed:
Hide labels on lg

yash-learner committed:
Refactor MedicationStatementGridRow layout for improved spacing and styling

yash-learner committed:
Use size

yash-learner committed:
Bump size

yash-learner committed:
Improve UI/UX

yash-learner committed:
Merge branch 'develop' into improve-ux-other-qc

yash-learner committed:
Add title and onBack props to ValueSetSelect

yash-learner committed:
Use size

yash-learner committed:
Add mobile drawer with new medication

yash-learner committed:
Update translations

yash-learner committed:
Merge branch 'develop' into improve-ux-other-qc

yash-learner committed:
Keep the notes inline

yash-learner committed:
WIP: Large refactor along with cursor

yash-learner committed:
Merge branch 'develop' into improve-ux-other-qc

yash-learner committed:
Always keep the note in the same line

yash-learner committed:
tailwind v4 field-sizing

yash-learner committed:
Improve note UI/UX and corresponding components

yash-learner committed:
Refactor ChoiceQuestion and QuestionInput components for consistent border styling

yash-learner committed:
Add drawer for allergy

yash-learner committed:
Some clean up

yash-learner committed:
Add icons to add and cancel buttons in EntitySelectionDrawer

yash-learner committed:
Merge branch 'develop' into improve-ux-other-qc

yash-learner committed:
Try fixing issue with keyboard and drawer layout on andriod

yash-learner committed:
Try fixing the double scroll

yash-learner committed:
Try fixing double scroll with keyboard

yash-learner committed:
Revert "Try fixing double scroll with keyboard"

This reverts commit 412f8de9bbaf37e21e1d24243fff88713e8e0037.

yash-learner committed:
Revert "Try fixing the double scroll"

This reverts commit dc55e12e8f1dadbd2c06d71db1deaa5fa799437d.

yash-learner committed:
Revert "Try fixing issue with keyboard and drawer layout on andriod"

This reverts commit 6bccd9959e3dc94e742fefeba42e11f6a7eaeca6.

yash-learner committed:
Try fixing issue with keyboard and drawer layout on andriod

yash-learner committed:
Try something

yash-learner committed:
Move button's to top

yash-learner committed:
More attempts to make it work

yash-learner committed:
Use just the drawer

yash-learner committed:
Try fix shrink when going back and selecting again

yash-learner committed:
Try fixing shrink

yash-learner committed:
Try fixing the shrink while going back adn selectign again

yash-learner committed:
Add some padding

yash-learner committed:
Use dvh

yash-learner committed:
Use the primary button

yash-learner committed:
Merge branch 'develop' into improve-ux-other-qc

yash-learner committed:
use text-sm

yash-learner committed:
Wrap up

- remove duplication check
- style changes

yash-learner committed:
Clean up

yash-learner committed:
Quick test

yash-learner committed:
Revert "Quick test"

This reverts commit 7cc3fc96fe7bfe3c020da74e86c72312bee14d89.

yash-learner committed:
Add some more padding

yash-learner committed:
replace the drawer with sheet

yash-learner committed:
Merge branch 'develop' into improve-ux-other-qc

yash-learner committed:
Wrap up

yash-learner committed:
Fix height

yash-learner committed:
Merge branch 'develop' into improve-ux-other-qc

nihal467 commented:
@yash-learner iam seeing commit beings pushed post the label, is it ready for testing

yash-learner commented:
> @yash-learner iam seeing commit beings pushed post the label, is it ready for testing


Yes @nihal467 it is ready for testing 

----

I pushed for fixing some merge conflicts and in one PR we had replaced drawer with sheet and so I did the same in my PR in similar context

amjithtitus09 commented:
@nihal467 Can you pick this up for testing?

amjithtitus09 committed:
Merge branch 'develop' into improve-ux-other-qc

yash-learner committed:
Merge branch 'develop' into improve-ux-other-qc

yash-learner committed:
Merge branch 'develop' into improve-ux-other-qc

yash-learner committed:
Let's see if this can fix collisions with virtual keyboard on mobile

yash-learner committed:
Merge branch 'develop' into improve-ux-other-qc

yash-learner committed:
Revert "Let's see if this can fix collisions with virtual keyboard on mobile"

This reverts commit de8e0fef3fa4fa8a8cf4ab082ee32b6d8574ae50.

amjithtitus09 commented:
@yash-learner 

1. While adding the first records itself, button shows up as Add "another" Symptom. Can you fix that?
2. When clicking on a valueset field in mobile, it used to open up the keyboard by default. Now it doesn't
3. When clicking on a text field inside the Add medication drawer, screen auto zooms in. Let's not do that.
4. Medication Request - Duration field is not shown in mobile for PRN. Isn't disabling enough?
5. Medication Request - Titrate dosage values aren't showing up on the collapsed card in mobile
6. "src/components/Medicine/MedicationAdministration/MedicineAdminForm.tsx:
Refactored for better mobile layout with improved spacing" - I see not code change or difference in Admin form other than the unit code 🤔
7. Consultation form - Admission recommended and Follow up advice dont't have same Notes styling

yash-learner committed:
Fix the placholder text

yash-learner committed:
Fix placeholder text for remaining components as well

yash-learner committed:
Fix keyboard autofocus and minior style enhancement

yash-learner committed:
Make it more disabled when it is PRN

yash-learner committed:
Show dosage range on collapsibe card

yash-learner committed:
Move the add note to next line for choice type question

yash-learner committed:
Merge branch 'develop' into improve-ux-other-qc

yash-learner committed:
Merge branch 'develop' into improve-ux-other-qc

yash-learner committed:
Merge branch 'develop' into improve-ux-other-qc

yash-learner committed:
Merge branch 'improve-ux-other-qc' of https://github.com/ohcnetwork/care_fe into improve-ux-other-qc

yash-learner committed:
Merge branch 'develop' into improve-ux-other-qc

yash-learner committed:
Merge branch 'develop' into improve-ux-other-qc

amjithtitus09 commented on src/components/Questionnaire/QuestionTypes/AllergyQuestion.tsx:
```diff
@@ -117,185 +126,351 @@ const AllergyTableRow = ({
+  const isInactiveStatus = allergy.clinical_status === "inactive";
+
+  const rowClassName = `group ${
+    allergy.verification_status === "entered_in_error"
```
Can we use isInactive here?

amjithtitus09 commented on src/components/Questionnaire/QuestionTypes/AllergyQuestion.tsx:
```diff
@@ -117,185 +126,351 @@ const AllergyTableRow = ({
+              })
+        }
+      >
+        <SelectValue placeholder={desktopLayout ? t("critical") : "Critical"} />
```
What is this doing? 🤔

yash-learner replied:
Cleaned up.

Probably missed cleaning up while accepting from cursor 😁

amjithtitus09 commented on src/components/Questionnaire/QuestionTypes/AllergyQuestion.tsx:
```diff
@@ -117,185 +126,351 @@ const AllergyTableRow = ({
+              })
+        }
+      >
+        <SelectValue placeholder={desktopLayout ? t("verify") : "Verify"} />
```
Same here

amjithtitus09 commented on src/components/Questionnaire/QuestionTypes/AllergyQuestion.tsx:
```diff
@@ -117,185 +126,351 @@ const AllergyTableRow = ({
+          last_occurrence: dateQueryString(date),
+        })
+      }
+      disabled={isInactive}
```
Shouldn't it check disabled as well?

amjithtitus09 replied:
Just realised isInactive checks disabled. A bit confusing IMO

amjithtitus09 commented on src/components/Questionnaire/QuestionTypes/AllergyQuestion.tsx:
```diff
@@ -117,185 +126,351 @@ const AllergyTableRow = ({
+                <Button
+                  variant="ghost"
+                  size="icon"
+                  disabled={isInactive}
```
Same here

amjithtitus09 replied:
Just realised isInactive checks disabled. A bit confusing IMO

amjithtitus09 commented on src/components/Questionnaire/QuestionTypes/AllergyQuestion.tsx:
```diff
@@ -117,185 +126,351 @@ const AllergyTableRow = ({
+  const isInactive =
+    allergy.verification_status === "entered_in_error" || disabled;
+
+  const notesDisabled = disabled;
```
This would end up being confusing right?

amjithtitus09 commented on src/components/Questionnaire/QuestionTypes/AllergyQuestion.tsx:
```diff
@@ -117,185 +126,351 @@ const AllergyTableRow = ({
+  const desktopLayout = useBreakpoints({ md: true, default: false });
+
+  const isInactive =
+    allergy.verification_status === "entered_in_error" || disabled;
```
Isn't it better to keep the disabled separate? What do you think?

yash-learner replied:
Now, I just reverted everything as it was https://github.com/ohcnetwork/care_fe/pull/11642/commits/781a029dfac0f0706b3fffeb9351af89bdebd5d8, previously

Jacobjeevan commented on src/components/Questionnaire/QuestionTypes/AllergyQuestion.tsx:
```diff
@@ -117,185 +126,351 @@ const AllergyTableRow = ({
+  const isResolved = allergy.clinical_status === "resolved";
+  const isInactiveStatus = allergy.clinical_status === "inactive";
+
+  const rowClassName = `group ${
```
can we switch to using cn?

Jacobjeevan commented on src/components/Questionnaire/QuestionTypes/AllergyQuestion.tsx:
```diff
@@ -117,185 +126,351 @@ const AllergyTableRow = ({
+  const handleNotesToggle = () => {
+    if (showNotes) {
+      setShowNotes(false);
+      onUpdate?.({ note: undefined });
```
onUpdate shouldn't be called on toggle, this will result in notes being updated again (when editing an existing item with a note).

Jacobjeevan commented on src/types/emr/medicationRequest.ts:
```diff
@@ -203,10 +238,10 @@ export const MEDICATION_REQUEST_TIMING_OPTIONS: Record<
         frequency: 2,
         period: 1,
-        period_unit: "d",
+        period_unit: "d" as UcumTimeUnitCode,
```
these shouldn't be necessary since you are specifying the type in lines 128 and 140.

Jacobjeevan commented on src/types/emr/medicationRequest.ts:
```diff
@@ -3,6 +3,10 @@ import { UserBareMinimum } from "@/components/Users/models";
 import { Code } from "@/types/questionnaire/code";
 
 export const DOSAGE_UNITS_CODES = [
+  // TODO: Are these smaller units required?
```
move this below under time units 😅  has nothing to do with dosage units

yash-learner replied:
> move this below under time units 😅 has nothing to do with dosage units

Yeah, I pasted in a wrong place

Jacobjeevan commented on src/components/Questionnaire/QuestionTypes/AllergyQuestion.tsx:
```diff
@@ -117,185 +126,351 @@ const AllergyTableRow = ({
+              })
+        }
+      >
+        <SelectValue placeholder={desktopLayout ? t("critical") : "Critical"} />
```
why 😅

Jacobjeevan commented on src/components/Questionnaire/QuestionTypes/AllergyQuestion.tsx:
```diff
@@ -117,185 +126,351 @@ const AllergyTableRow = ({
+    <Select
+      value={allergy.verification_status}
+      onValueChange={(value) => {
+        if (value in ALLERGY_VERIFICATION_STATUS) {
```
why do we need this? the only available options are ones specified in allergy_verification_status.

yash-learner replied:
Fixed it https://github.com/ohcnetwork/care_fe/pull/11642/commits/dad1533b3119b484942723cd88c80ac94a3cf307

Jacobjeevan reviewed (changes requested):
Make sure to double check against latest 🤔  I haven't reviewed the whole thing yet.

Like the onUpdate for notes - I had specifically requested Aditya to fix in another PR (merged).

yash-learner committed:
Merge branch 'develop' into improve-ux-other-qc

yash-learner committed:
Fix review comments

- Move comment to right place
- Remove unnecessary type conversion

yash-learner committed:
Use Cn, remove unecessary code

yash-learner committed:
Just keep everything as it was in behaviour

yash-learner committed:
Merge branch 'develop' into improve-ux-other-qc

yash-learner committed:
Fix spacing and note

yash-learner committed:
FIx some things back

yash-learner committed:
Bring back changes made in PR#11621

yash-learner committed:
Merge branch 'develop' into improve-ux-other-qc

yash-learner commented on src/components/Questionnaire/QuestionTypes/AllergyQuestion.tsx:
```diff
@@ -429,229 +650,195 @@ export function AllergyQuestion({
-                {allergy.note !== undefined && (
-                  <div>
-                    <Label className="text-xs text-gray-500">
-                      {t("notes")}
-                    </Label>
-                    <Input
-                      type="text"
-                      placeholder="Add notes about the allergy..."
-                      value={allergy.note ?? ""}
-                      onChange={(e) =>
-                        handleUpdateAllergy(index, { note: e.target.value })
-                      }
-                      disabled={disabled}
-                      className="mt-1"
-                    />
-                  </div>
-                )}
-              </div>
+                      </div>
+                    </CardHeader>
+                  </CollapsibleTrigger>
+                  <CollapsibleContent>
+                    <CardContent
+                      className={cn(
+                        "p-3 pt-2 space-y-3 rounded-lg bg-gray-50",
+                        allergy.verification_status === "entered_in_error" &&
+                          "pointer-events-none",
+                      )}
+                    >
+                      <AllergyTableRow
+                        allergy={allergy}
+                        disabled={disabled}
+                        onUpdate={(updates) =>
+                          handleUpdateAllergy(index, updates)
+                        }
+                        onRemove={() => handleRemoveAllergy(index)}
+                      />
+                    </CardContent>
+                  </CollapsibleContent>
+                </Card>
```
@coderabbitai I didn't get, please elaborate

yash-learner replied:
@coderabbitai what refactor suggestion you are giving, please elaborate that

yash-learner replied:
@coderabbitai okay

I just got confused by `Actionable comments posted: 1` made by you

yash-learner committed:
Clean up

rithviknishad commented on src/components/Questionnaire/QuestionTypes/DiagnosisQuestion.tsx:
```diff
@@ -547,6 +538,10 @@ export function DiagnosisQuestion({
 
+  const addDiagnosisPlaceholder = t(
+    sortedDiagnoses.length === 0 ? "add_diagnosis" : "add_another_diagnosis",
+  );
```
https://www.i18next.com/translation-function/plurals

rithviknishad commented on src/components/Questionnaire/QuestionTypes/MedicationRequestQuestion.tsx:
```diff
@@ -277,6 +321,34 @@ export function MedicationRequestQuestion({
+
+  const addMedicationPlaceholder = t(
+    medications.length === 0 ? "add_medication" : "add_another_medication",
+  );
```
https://www.i18next.com/translation-function/plurals

rithviknishad commented on src/components/Questionnaire/QuestionTypes/NoteToggle.tsx:
```diff
@@ -0,0 +1,61 @@
+        className="h-full w-28 text-sm font-normal text-gray-700 hover:text-gray-900 gap-1 bg-gray-50 rounded-none"
+      >
+        {hasNotes ? (
+          <div className="w-1.5 h-1.5 rounded-full bg-orange-400" />
```
nitpick:
```suggestion
          <div className="size-1.5 rounded-full bg-orange-400" />
```

rithviknishad commented on src/components/Questionnaire/QuestionTypes/SymptomQuestion.tsx:
```diff
@@ -704,18 +700,15 @@ export function SymptomQuestion({
     </div>
   );
 
+  const addSymptomPlaceholder = t(
+    symptoms.length === 0 ? "add_symptom" : "add_another_symptom",
+  );
+
```
https://www.i18next.com/translation-function/plurals

rithviknishad commented on src/types/emr/medicationRequest.ts:
```diff
@@ -35,11 +35,31 @@ export const UCUM_TIME_UNITS = [
-  "wk",
-  "mo",
-  "a",
+  {
+    code: "h",
+    display: "Hour",
+    system: "http://unitsofmeasure.org",
+  },
+  {
+    code: "d",
+    display: "Day",
+    system: "http://unitsofmeasure.org",
+  },
+  {
+    code: "wk",
+    display: "Week",
+    system: "http://unitsofmeasure.org",
+  },
+  {
+    code: "mo",
+    display: "Month",
+    system: "http://unitsofmeasure.org",
+  },
+  {
+    code: "a",
+    display: "Year",
+    system: "http://unitsofmeasure.org",
+  },
```
don't we need translations for these displays? 🤔

rithviknishad reviewed (changes requested):
only did a quick pass

yash-learner committed:
WIP: add a sheet for multivalueselect

yash-learner committed:
Fix the height

yash-learner committed:
Fix height

yash-learner committed:
Add the scrollarea for dosage on Mobile as popover with sheet having issues with z index

yash-learner committed:
Merge branch 'develop' into improve-ux-other-qc

yash-learner committed:
Fix: Add max height and overflow for SelectContent in MedicationRequestGridRow

yash-learner committed:
Improve height adjustment

yash-learner committed:
Move the tabs to top so that we will have more space on real devices with keyboard

yash-learner committed:
Revert "Improve height adjustment"

This reverts commit ba15efaff28b2bdacba16dd9bd24b715a6fe562d.

yash-learner committed:
Merge branch 'develop' into improve-ux-other-qc

yash-learner committed:
Use i18n plural syntax

yash-learner committed:
Use size

yash-learner committed:
Add translation keys for dosage unit display

yash-learner committed:
Add translation keys for time units

nihal467 commented:
@yash-learner can you clear the merge conflict

yash-learner committed:
Merge remote-tracking branch 'upstream/develop' into improve-ux-other-qc

yash-learner committed:
Revert "Add the scrollarea for dosage on Mobile as popover with sheet having issues with z index"

This reverts commit 9d75cf995619d10682ac161a25aa8e4d1683a18d.

yash-learner commented:
> @yash-learner can you clear the merge conflict

@nihal467 fixed it

and also removed code because popover things got fixed with react 19 upgrade

yash-learner committed:
Merge branch 'develop' into improve-ux-other-qc

yash-learner committed:
Merge branch 'develop' into improve-ux-other-qc

nihal467 committed:
Merge branch 'develop' into improve-ux-other-qc

nihal467 commented:
@yash-learner Cypress reported a failure due to the behavior change you made to the dosage field. Could you please fix it? The dosage field should retain the same behavior as it has in our current develop branch.

yash-learner committed:
Add accidently removed value

fix cypress

yash-learner committed:
Merge branch 'develop' into improve-ux-other-qc

yash-learner committed:
Merge branch 'develop' into improve-ux-other-qc

yash-learner committed:
Use grid layout for reposnsiveness of duration

yash-learner committed:
Revert "Add translation keys for dosage unit display"

This reverts commit c402c721db598e840d7895565cfd856e085e7110.

yash-learner committed:
Fix case

yash-learner commented:
Thanks @nihal467 for finding them 🙌



> * When we submit a Medication Request, and view the preview in the medication section, the units are visible without 

I reverted this I assumed it is working similar to duration but not, so fixing it will lead to more changes as this PR already has to many changes, I just reverted it out.

> * [ ]  the text area used by the number in duration is different when the units are day and days, where the design is becoming inconsistent.

I fixed it now

![image](https://github.com/user-attachments/assets/07a51722-f205-44fb-b317-d76938e22183)


> @yash-learner The newly identified bugs have appeared in the same area where Cypress initially flagged a functionality issue. Since repeated functionality problems are being observed in your UX-related PRs, Request you to perform a thorough self-review of both functionality and design before submitting your PR for review

These tiny issues came after adding translations in the last minute, I should have checked it. 😁

I will do a full check and then request again, sorry for inconvience @nihal467 


 CC: @amjithtitus09

gigincg commented:
Closing the PR as discussed. Let's create a new branch to implement the new EntitySelectionDrawer
